src/sessions/p2p_session.rs

```diff
@@ -268,14 +273,17 @@ impl<T: Config> P2PSession<T> {
 
         // session is not running and synchronized
         if self.state != SessionState::Running {
+            trace!("Session not synchronized; returning error");
             return Err(GgrsError::NotSynchronized);
         }
 
         // check if input for all local players is queued
         for handle in self.player_reg.local_player_handles() {
             if !self.local_inputs.contains_key(&handle) {
                 return Err(GgrsError::InvalidRequest {
-                    info: "Missing local input while calling advance_frame().".to_owned(),
+                    info: format!(
+                        "Missing local input for handle {handle} while calling advance_frame()."
```

Comment:
Here's the non-logging change: when the user didn't provide input for a handle, include the handle with missing input in the error message. I suppose technically it's a breaking change, but it would be super, super surprising if anyone was actually relying on parsing the error messages. (If anyone does complain, we can fix it by adding dedicated error types for each possible error instead.)

Anyway, this is totally not logging related, but still kind of matching the theme of improving debuggability of this library. Still, let me know if you want me to split it out into its own trivial PR for the sake of having PRs be single-purpose.

Reply:
We're still 0.x, lets break the API :)

Reply:
Well, as a Rust codebase, we are officially not allowed to ever release version 1.0 ;)

(sure, sounds good)